Display multi seq on ui

MImranAsghar opened the pull request:
Adds multi seq UI display for https://github.com/ProjectMirador/mirador/issues/2949
  - Shows sequences as collapsible buttons in sidebar
Updates sequences selector in master by adding getSequenceCanvases selector

Before

<img width="1440" alt="Before" src="https://user-images.githubusercontent.com/12305084/96768809-4651ca00-13ac-11eb-8e33-b272846d55e8.png">

After UI update

<img width="1440" alt="After_1" src="https://user-images.githubusercontent.com/12305084/96768866-579ad680-13ac-11eb-97f0-98c269cb2702.png">
<img width="1440" alt="After_2" src="https://user-images.githubusercontent.com/12305084/96768871-58cc0380-13ac-11eb-9718-5aec1162623f.png">
<img width="1440" alt="After_3" src="https://user-images.githubusercontent.com/12305084/96768873-59649a00-13ac-11eb-915a-0ac8e3082f3a.png">

mejackreed commented:
Thanks @MImranAsghar . Are there any public manfiests that you could link to here where we could look at this feature? It might help make evaluation of the design of this PR easier.

MImranAsghar commented:
Hi Jack, here is one I use often https://www.e-codices.unifr.ch/metadata/iiif/gau-Fragment/manifest.json

mejackreed reviewed (changes requested):
@MImranAsghar thanks so much for submitting this PR. We especially like how you added tests here for the work you added!

While reviewing with @ggeisler there is some concern about the collapsible panels and were wondering if you would be open to either:
 - adapting this pull request to use the designed select dropdown for sequences
 - open to pull requests that implement this feature

An example of this here:

![Screen Shot 2020-10-21 at 5 27 17 PM](https://user-images.githubusercontent.com/1656824/96800500-b23f2d00-13c2-11eb-9429-6269e41975f4.png)

There is a desire to keep the toggle buttons to select index/thumbnails but would preference sequence selection from the select dropdown.

MImranAsghar commented:
Hi @mejackreed , I see, no worries :) For the select dropdowns on ui, I assume it was done before here https://github.com/ProjectMirador/mirador/compare/2949-multi-seq#diff-dd9bcb599eb3ffafce45ff153b58fa4bbcc455502e09c255fdf43c91f4b31c91R98 but seems like this would have to be done differently now because WindowSideBarCanvasPanel code in master has changed from when it was done? If so, I can adapt this pull request (https://github.com/ProjectMirador/mirador/pull/3299) I made yesterday for collapsible panels to display select dropdown instead.

mejackreed commented:
@MImranAsghar yes for sure some things have changed. However, I think the seam might be similar implemented as part of the `titleControls` here: https://github.com/ProjectMirador/mirador/blob/master/src/components/WindowSideBarCanvasPanel.js#L86-L100

We should maybe just add above (or maybe within a fragment) the formcontrol selection:

https://github.com/ProjectMirador/mirador/compare/2949-multi-seq#diff-dd9bcb599eb3ffafce45ff153b58fa4bbcc455502e09c255fdf43c91f4b31c91R98-R128

MImranAsghar commented:
@mejackreed Ah okay true, makes sense 👍 I guess i will move the formcontrol selection in that case. I will also add some tests for that.

MImranAsghar commented:
Hi @mejackreed , so I have updated the select dropdown UI from (https://github.com/ProjectMirador/mirador/tree/2949-multi-seq). Updated it to work with the latest master. I have created a fresh pull request https://github.com/ProjectMirador/mirador/pull/3304

mejackreed commented:
Superceded by #3304
